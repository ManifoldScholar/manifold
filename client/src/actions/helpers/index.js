export { default as createApiAction } from "./createApiAction";
export { default as commonActions } from "./commonActions";
