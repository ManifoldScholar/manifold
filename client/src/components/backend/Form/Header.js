import React, { Component } from "react";
import PropTypes from "prop-types";

export default class FormHeader extends Component {
  static displayName = "Form.Header";

  static propTypes = {
    label: PropTypes.string.isRequired
  };

  render() {
    return (
      <header className="form-header">
        <h2>{this.props.label}</h2>
      </header>
    );
  }
}
