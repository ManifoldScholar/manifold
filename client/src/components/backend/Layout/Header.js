import React, { Component } from "react";
import PropTypes from "prop-types";
import {
  HeaderNotifications,
  UIPanel,
  UserMenuBody,
  UserMenuButton,
  PressLogo
} from "components/global";
import { Link, NavLink } from "react-router-dom";
import startsWith from "lodash/startsWith";
import get from "lodash/get";
import lh from "helpers/linkHandler";

export default class LayoutHeader extends Component {
  static displayName = "Layout.Header";

  static propTypes = {
    visibility: PropTypes.object,
    location: PropTypes.object,
    authentication: PropTypes.object,
    notifications: PropTypes.object,
    commonActions: PropTypes.object,
    settings: PropTypes.object
  };

  isProjects(match, location) {
    if (!match) {
      return false;
    }
    const { pathname } = location;
    if (pathname === "/backend") return true;
    if (startsWith(pathname, "/backend/project")) return true;
    if (startsWith(pathname, "/backend/resource")) return true;
    if (startsWith(pathname, "/backend/text")) return true;
    return false;
  }

  isSubjects(match, location) {
    if (!match) {
      return false;
    }
    const { pathname } = location;
    if (startsWith(pathname, "/backend/subjects")) return true;
  }

  render() {
    return (
      <header className={"header-app dark"}>
        <div className="header-container">
          <Link to={lh.link("frontend")} className="logo">
            <PressLogo
              url={get(this.props.settings, "attributes.pressLogoStyles.small")}
            />
          </Link>
          <nav className="text-nav">
            <ul>
              <li>
                <NavLink
                  isActive={this.isProjects}
                  activeClassName="active"
                  to={lh.link("backend")}
                >
                  {"Projects"}
                </NavLink>
              </li>
              <li>
                <NavLink activeClassName="active" to={lh.link("backendPeople")}>
                  {"People"}
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName="active"
                  to={lh.link("backendContent")}
                >
                  {"Content"}
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName="active"
                  to={lh.link("backendSettings")}
                >
                  {"Settings"}
                </NavLink>
              </li>
            </ul>
          </nav>

          <nav className="menu-dropdowns">
            <ul>
              <li className="show-60">
                <Link className="button-mode" to={lh.link("frontend")}>
                  Exit Admin Mode
                </Link>
              </li>
              <li>
                <UserMenuButton
                  authentication={this.props.authentication}
                  active={this.props.visibility.uiPanels.user}
                  showLoginOverlay={
                    this.props.commonActions.toggleSignInUpOverlay
                  }
                  toggleUserMenu={this.props.commonActions.toggleUserPanel}
                />
                <UIPanel
                  id="user"
                  visibility={this.props.visibility.uiPanels}
                  bodyComponent={UserMenuBody}
                  showLoginOverlay={
                    this.props.commonActions.toggleSignInUpOverlay
                  }
                  startLogout={this.props.commonActions.logout}
                  hideUserMenu={this.props.commonActions.toggleUserPanel}
                />
              </li>
            </ul>
          </nav>
        </div>

        <div className="header-border" />

        <HeaderNotifications
          notifications={this.props.notifications}
          addNotification={this.props.commonActions.addNotification}
          removeNotification={this.props.commonActions.removeNotification}
          removeAllNotifications={this.props.commonActions.clearNotifications}
        />
      </header>
    );
  }
}
