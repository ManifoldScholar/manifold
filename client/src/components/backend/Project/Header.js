import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';

export default class ProjectHeader extends Component {

  static displayName = "Project.Header";

  static propTypes = {
    project: PropTypes.object
  };

  render() {
    const project = this.props.project;
    const attr = project.attributes;
    return (
      <section className="bg-neutral95">
        <div className="container flush">
          <Link to={'/backend/'} className="back-link-primary">
            <i className="manicon manicon-arrow-left"></i>
            Back to: <span>ALL PROJECTS</span>
          </Link>
          <header className="project-header">
            <figure>
              <i className="manicon manicon-project-placeholder"></i>
            </figure>
            <div className="project-title">
              <h1>
                {attr.title}
                <span className="subtitle">
                  {attr.subtitle}
                </span>
              </h1>
              <div className="project-utility">
                <button className="button-bare-primary">
                  Preview <i className="manicon manicon-eye-outline"></i>
                </button>
                <button className="button-bare-primary">
                  Duplicate <i className="manicon manicon-check-double"></i>
                </button>
                <button className="button-bare-primary">
                  Delete <i className="manicon manicon-trashcan"></i>
                </button>
              </div>
            </div>
          </header>
        </div>
      </section>
    );
  }
}
