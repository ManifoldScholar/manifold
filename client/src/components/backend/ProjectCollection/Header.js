import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { Navigation } from "components/backend";
import { Link } from "react-router-dom";
import labelId from "helpers/labelId";
import lh from "helpers/linkHandler";

export default class ProjectCollectionHeader extends PureComponent {
  static displayName = "ProjectCollection.Header";

  static propTypes = {
    projectCollection: PropTypes.object
  };

  static defaultProps = {
    sortId: labelId("project-collection-header-")
  };

  renderUtility = projectCollection => {
    const smart = projectCollection.attributes.smart;

    return (
      <div>
        <Link
          to={lh.link("backendProjectCollectionSettings", projectCollection.id)}
          className="button-bare-primary"
        >
          <i className="manicon manicon-settings" aria-hidden="true" />
          Settings
        </Link>
        {!smart ? (
          <Link
            to={lh.link(
              "backendProjectCollectionManageProjects",
              projectCollection.id
            )}
            className="button-bare-primary"
          >
            <i className="manicon manicon-stack" aria-hidden="true" />
            Manage Projects
          </Link>
        ) : null}
      </div>
    );
  };

  renderPlaceholder() {
    return (
      <section className="backend-header">
        <div className="wrapper">
          <header className="entity-header-primary instructional">
            Please select a Project Collection from the list to edit its content
            and settings.
          </header>
        </div>
      </section>
    );
  }

  render() {
    const { projectCollection } = this.props;
    if (!projectCollection) return this.renderPlaceholder();

    const iconName = projectCollection.attributes.smart
      ? `collection-smart`
      : `collection-manual`;

    return (
      <Navigation.DetailHeader
        iconName={iconName}
        title={projectCollection.attributes.title}
        utility={this.renderUtility(projectCollection)}
        backUrl={lh.link("backendProjectCollections")}
      />
    );
  }
}
