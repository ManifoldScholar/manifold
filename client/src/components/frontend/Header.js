import React, { Component, PropTypes } from 'react';
import {Link} from 'react-router';

export default class Header extends Component {

  static propTypes = {
    texts: PropTypes.object
  };

  render() {
    return (
      <header className={'header-browse'}>
        <Link to={'/browse'} className="logo">
          <figure>
            <i className="manicon manicon-manifold-logo"></i>
            <span className="screen-reader-text">
              {'Manifold Logo: Click to return to the browse page'}
            </span>
          </figure>
        </Link>
        <nav className="text-nav">
          <ul>
            <li className={this.props.location.pathname == '/browse/' ? 'active' : ''}>
              <Link to={`/browse/`}>
                Browse
              </Link>
            </li>
            <li className={this.props.location.pathname == '/browse/following/' ? 'active' : ''}>
              <Link to={`/browse/following/`}>
                Following
              </Link>
            </li>
          </ul>
        </nav>

        <nav className="widget-nav">
          <button className="button-bare-icon button-magnify">
            <i className="manicon manicon-magnify"></i>
            <span className="screen-reader-text">{'Click to search Manifold library'}</span>
          </button>
          <Link to={'/browse/login'}>
            <button className="button-avatar">
              <i className="manicon manicon-person"></i>
              <span className="screen-reader-text">{'Click to open user settings'}</span>
            </button>
          </Link>
        </nav>
      </header>
    );
  }
}
