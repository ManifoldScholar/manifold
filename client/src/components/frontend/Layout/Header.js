import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import {
  HeaderNotifications,
  UIPanel,
  UserMenuBody,
  UserMenuButton,
  PressLogo
} from "components/global";
import { Link } from "react-router-dom";
import startsWith from "lodash/startsWith";
import get from "lodash/get";
import HigherOrder from "containers/global/HigherOrder";
import lh from "helpers/linkHandler";

export default class LayoutHeader extends PureComponent {
  static displayName = "Layout.Header";

  static propTypes = {
    visibility: PropTypes.object,
    location: PropTypes.object,
    authentication: PropTypes.object,
    notifications: PropTypes.object,
    commonActions: PropTypes.object,
    settings: PropTypes.object,
    pages: PropTypes.array
  };

  static defaultProps = {
    pages: []
  };

  visiblePages(props) {
    if (!props.pages) return [];
    return props.pages.filter(p => {
      return p.attributes.showInHeader && !p.attributes.hidden;
    });
  }

  render() {
    const path = this.props.location.pathname;
    const projectsActive = path === "/" || startsWith(path, "/project");
    const followingActive = startsWith(path, "/following");

    return (
      <header className={"header-app"}>
        <div className="header-container">
          <Link to={lh.link("frontend")} className="logo">
            <PressLogo
              url={get(
                this.props.settings,
                "attributes.pressLogoStyles.original"
              )}
              styles={get(this.props.settings, "attributes.theme.logoStyles")}
            />
          </Link>
          {/* Use show-50 utility class to hide text-nav on mobile */}
          <nav className="text-nav show-50">
            <ul>
              <li className={projectsActive ? "active" : ""}>
                <Link to={lh.link("frontend")}>{"Projects"}</Link>
              </li>
              <HigherOrder.RequireKind requiredKind="any">
                <li className={followingActive ? "active" : ""}>
                  <Link to={lh.link("frontendFollowing")}>{"Following"}</Link>
                </li>
              </HigherOrder.RequireKind>
              {this.visiblePages(this.props).map(page => {
                const url = lh.link("frontendPage", page.attributes.slug);
                return (
                  <li key={page.id} className={path === url ? "active" : ""}>
                    <Link to={url}>
                      {page.attributes.navTitle || page.attributes.title}
                    </Link>
                  </li>
                );
              })}
            </ul>
          </nav>

          <nav className="menu-dropdowns">
            <ul>
              <HigherOrder.RequireKind requiredKind={"admin"}>
                <li>
                  <Link className="button-mode" to={lh.link("backend")}>
                    Admin Mode
                  </Link>
                </li>
              </HigherOrder.RequireKind>

              {/*
                Hiding search markup until functionality is available
                <li>
                  <SearchMenuButton
                    toggleSearchMenu={this.props.commonActions.toggleSearchPanel}
                    active={this.props.visibility.uiPanels.search}
                  />
                  <UIPanel
                    id="search"
                    visibility={this.props.visibility.uiPanels}
                    bodyComponent={SearchMenuBody}
                  />
                </li>
              */}
              <li>
                <UserMenuButton
                  authentication={this.props.authentication}
                  active={this.props.visibility.uiPanels.user}
                  showLoginOverlay={
                    this.props.commonActions.toggleSignInUpOverlay
                  }
                  toggleUserMenu={this.props.commonActions.toggleUserPanel}
                />
                <UIPanel
                  id="user"
                  visibility={this.props.visibility.uiPanels}
                  bodyComponent={UserMenuBody}
                  showLoginOverlay={
                    this.props.commonActions.toggleSignInUpOverlay
                  }
                  startLogout={this.props.commonActions.logout}
                  hideUserMenu={this.props.commonActions.toggleUserPanel}
                />
              </li>
            </ul>
          </nav>
        </div>

        <div className="header-border" />

        <HeaderNotifications
          notifications={this.props.notifications}
          addNotification={this.props.commonActions.addNotification}
          removeNotification={this.props.commonActions.removeNotification}
          removeAllNotifications={this.props.commonActions.clearNotifications}
        />
      </header>
    );
  }
}
