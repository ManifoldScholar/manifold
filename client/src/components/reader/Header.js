import React, { Component, PropTypes } from 'react';
import { TocDrawer } from './';
import { Link } from 'react-router';

import classNames from 'classnames';

export default class Header extends Component {

  static propTypes = {
    text: PropTypes.object,
    tocVisible: PropTypes.bool,
    toggleTocDrawer: PropTypes.func,
    hideTocDrawer: PropTypes.func,
  };

  render = () => {
    const buttonIndexClass = classNames({
      'button-index': true,
      'active': this.props.tocVisible,
    });
    return (
        <header className="header-reader">
          <Link to={`/browse/project/${this.props.text.relationships.project.data.id}`} >
            <button className="button-close" >
              <i className="manicon manicon-x"></i>
                <span className="screen-reader-text">
                  {'Click to close reader'}
                </span>
            </button>
          </Link>
          <button className={buttonIndexClass} onClick={this.props.toggleTocDrawer}>
            {'Contents'}<i className="manicon manicon-caret-down"></i>
          </button>
          <h2 className="title">
            {this.props.text.attributes.title}
          </h2>
          <TocDrawer text={this.props.text} visible={this.props.tocVisible} hideTocDrawer={this.props.hideTocDrawer} />
        </header>
    );
  };
}
