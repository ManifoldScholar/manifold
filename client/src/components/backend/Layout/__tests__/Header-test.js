import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';
import build from 'test/fixtures/build';
import { wrapWithRouter, renderWithRouter } from 'test/helpers/routing';

describe("Backend.Layout.Header component", () => {

  const settings = build.entity.settings();
  const toggleOverlayMock = jest.fn();
  const togglerUserPanelMock = jest.fn();
  const startLogoutMock = jest.fn();


  it('renders correctly', () => {
    const component = renderer.create(
      wrapWithRouter(
        <Header
          authentication={{ authenticated: false }}
          notifications={{ notifications: {} }}
          location={{}}
          visibility={{ uiPanels: {} }}
          settings={settings}
          commonActions={{
            toggleSignInUpOverlay: toggleOverlayMock,
            toggleUserPanel: togglerUserPanelMock,
            logout: startLogoutMock
          }}
        />
      )
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
