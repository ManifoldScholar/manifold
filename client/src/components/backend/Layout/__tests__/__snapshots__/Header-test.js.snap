// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backend.Layout.Header component renders correctly 1`] = `
<header
  className="header-app dark"
>
  <div
    className="header-container"
  >
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      <figure
        className=""
      >
        <div>
          <i
            className="manicon manicon-manifold-logo"
          />
          <span
            className="screen-reader-text"
          >
            Manifold Logo: Click to return to the browse page
          </span>
        </div>
      </figure>
    </a>
    <nav
      className="text-nav"
    >
      <ul>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/backend"
            onClick={[Function]}
            style={undefined}
          >
            Projects
          </a>
        </li>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/backend/people"
            onClick={[Function]}
            style={undefined}
          >
            People
          </a>
        </li>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/backend/content"
            onClick={[Function]}
            style={undefined}
          >
            Content
          </a>
        </li>
        <li>
          <a
            aria-current={false}
            className={undefined}
            href="/backend/settings"
            onClick={[Function]}
            style={undefined}
          >
            Settings
          </a>
        </li>
      </ul>
    </nav>
    <nav
      className="menu-dropdowns"
    >
      <ul>
        <li
          className="show-60"
        >
          <a
            className="button-mode"
            href="/"
            onClick={[Function]}
          >
            Exit Admin Mode
          </a>
        </li>
        <li>
          <button
            className="button-avatar"
            onClick={[Function]}
          >
            <span
              className="screen-reader-text"
            >
              Click to login or open user settings
            </span>
            <figure
              className="avatar"
              style={undefined}
            >
              <i
                className="manicon manicon-person"
              />
            </figure>
          </button>
          <div
            className="panel-hidden"
          >
            <nav
              className="user-menu menu-hidden"
            >
              <i
                className="tail"
              />
              <ul>
                <li>
                  <button
                    onClick={[Function]}
                  >
                    <i
                      className="manicon manicon-person-pencil-simple"
                    />
                    Edit Profile
                  </button>
                </li>
                <li>
                  <button
                    onClick={[Function]}
                  >
                    <i
                      className="manicon manicon-circle-arrow-out-right"
                    />
                    Logout
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="header-border"
  />
  <section
    className="header-notifications-container"
  >
    <div
      className="header-notifications-list"
    >
      <span />
    </div>
  </section>
</header>
`;
